Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -21,6 +21,13 @@ class ChannelDTO: NSManagedObject {
     @NSManaged var updatedAt: DBDate
     @NSManaged var lastMessageAt: DBDate?
 
+    // Index of the channel in the channel list query response.
+    // It is used for tie breaker when sorting by possibly equal values.
+    // Note: When the channel is part of multiple channel list controllers,
+    // it might still cause some movement because the property is shared.
+    // Recommendation: use at least one sorting parameter with unique values.
+    @NSManaged var sortIndex: Int
```

Comment:
another option can be to keep a map of the queries and the indexes, but I guess it's not worth it

Reply:
I considered it but it felt like it gets too complex and also sort descriptor wants a key-path.